### ISSUE TITLE
Output field order inconsistent with the input

### ISSUE THREAD

**Issue by Shawyeok** [eiiches/jackson-jq#81]
```shell
$ jackson-jq -n '{a:{depth:1,b:{depth:2}}}'
{
  "a" : {
    "b" : {
      "depth" : 2
    },
    "depth" : 1
  }
}

$ jq -n '{a:{depth:1,b:{depth:2}}}'
{
  "a": {
    "depth": 1,
    "b": {
      "depth": 2
    }
  }
}
```

Problem may be here: 
```java
// method: net.thisptr.jackson.jq.internal.tree.ObjectConstruction#apply
public void apply(final Scope scope, final JsonNode in, final Path ipath, final PathOutput output, final boolean requirePath) throws JsonQueryException {
  final Map<String, JsonNode> tmp = new HashMap<>();
  applyRecursive(scope, in, output, fields, tmp);
}
```
Should use LinkedHashMap instead, and testcase seem doesn't care field order.
[net/thisptr/jackson/jq/internal/misc/JsonNodeComparator.java#L94-L94](https://github.com/eiiches/jackson-jq/blob/develop/1.x/jackson-jq/src/main/java/net/thisptr/jackson/jq/internal/misc/JsonNodeComparator.java#L94-L94)

**Comment by eiiches** [comment]
Thanks for reporting the issue. It was a design decision not to maintain the order or the keys in a JSON object because at that time I had (and still have) an opinion that we shouldn't rely on the order of the keys in JSON.

That said, I also think the compatibility with jq is more important now. We can't fix this in any time soon because we probably have to rewrite dozens of places throughout the codebase where objects are created not only in the ObjectConstruction class, but I think we can start by replacing the HashMap there with a LinkedHashMap. Are you willing to submit a PR with the change?

Also, I realized I didn't mention this in README. I just added a note there.
